test/test_testing.py

```diff
@@ -1398,6 +1366,180 @@ def test_matching_per_channel(self):
+
+    @supported_dtypes
+    @parametrize("noncontiguous", [False, True])
+    @parametrize("shape", [tuple(), (0,), (1,), (1, 1), (2,), (2, 3), (8, 16, 32)])
+    def test_noncontiguous(self, dtype, device, noncontiguous, shape):
+        numel = functools.reduce(lambda a, b: a * b, shape, 1)
+
+        t = torch.testing.make_tensor(shape, dtype=dtype, device=device, noncontiguous=noncontiguous)
+        self.assertEqual(t.is_contiguous(), not noncontiguous or numel < 2)
+
+    @supported_dtypes
+    @parametrize(
+        "memory_format_and_shape",
+        [
+            (None, (2, 3, 4)),
+            (torch.contiguous_format, (2, 3, 4)),
+            (torch.channels_last, (2, 3, 4, 5)),
+            (torch.channels_last_3d, (2, 3, 4, 5, 6)),
+            (torch.preserve_format, (2, 3, 4)),
+        ],
+    )
+    def test_memory_format(self, dtype, device, memory_format_and_shape):
+        memory_format, shape = memory_format_and_shape
+
+        t = torch.testing.make_tensor(shape, dtype=dtype, device=device, memory_format=memory_format)
+
+        self.assertTrue(
+            t.is_contiguous(memory_format=torch.contiguous_format if memory_format is None else memory_format)
+        )
+
+    @supported_dtypes
+    def test_noncontiguous_memory_format(self, dtype, device):
+        with self.assertRaises(AssertionError):
+            torch.testing.make_tensor(
+                (2, 3, 4, 5),
+                dtype=dtype,
+                device=device,
+                noncontiguous=True,
+                memory_format=torch.channels_last,
+            )
+
+    @supported_dtypes
+    def test_exclude_zero(self, dtype, device):
+        t = torch.testing.make_tensor(10_000, dtype=dtype, device=device, exclude_zero=True, low=-1, high=2)
+
+        self.assertTrue((t != 0).all())
+
+    @supported_dtypes
+    def test_low_high_smoke(self, dtype, device):
+        low_inclusive, high_exclusive = 0, 2
+
+        t = torch.testing.make_tensor(10_000, dtype=dtype, device=device, low=low_inclusive, high=high_exclusive)
+        if dtype.is_complex:
+            t = torch.view_as_real(t)
+
+        self.assertTrue(((t >= low_inclusive) & (t < high_exclusive)).all())
+
+    @supported_dtypes
+    def test_low_high_default_smoke(self, dtype, device):
+        low_inclusive, high_exclusive = {
+            torch.bool: (0, 2),
+            torch.uint8: (0, 10),
+            **{
+                signed_integral_dtype: (-9, 10)
+                for signed_integral_dtype in [
+                    torch.int8,
+                    torch.int16,
+                    torch.int32,
+                    torch.int64,
+                ]
+            },
+        }.get(dtype, (-9, 9))
+
+        t = torch.testing.make_tensor(10_000, dtype=dtype, device=device, low=low_inclusive, high=high_exclusive)
+        if dtype.is_complex:
+            t = torch.view_as_real(t)
+
+        self.assertTrue(((t >= low_inclusive) & (t < high_exclusive)).all())
```

Comment:
```
_________ TestMakeTensorCUDA.test_low_high_default_smoke_cuda_bfloat16 _________
Traceback (most recent call last):
  File "/var/lib/jenkins/workspace/test/test_testing.py", line 1481, in test_low_high_default_smoke
    self.assertTrue(((t >= low_inclusive) & (t < high_exclusive)).all())
  File "/opt/conda/envs/py_3.10/lib/python3.10/unittest/case.py", line 687, in assertTrue
    raise self.failureException(msg)
AssertionError: tensor(False, device='cuda:0') is not true
```

The test failures look real -- maybe there's a bug with uniform for bfloat16 types?